### PR TITLE
OCPBUGS-11292: Allow hub template lookup function for ManagedCluster resource

### PR DESCRIPTION
**Pull request by Missxiaoguo**
This PR depends on https://github.com/openshift-kni/cluster-group-upgrades-operator/pull/524.

/cc @jc-rh

### CODE REVIEW

File: controllers/utils/policy_template_test.go [hunk 1]
```diff
@@ -52,6 +52,9 @@ spec:
         - '{{hub (printf "%s-name" .ManagedClusterName | fromConfigMap "ztp-common" "common-cm") | base64enc hub}}'
         - '{{hub (printf "%s-name" .ManagedClusterName | fromConfigMap "ztp-common" "common-cm") | toInt hub}}-{{hub fromConfigMap "ztp-common" "common-cm" "common-key" | toInt hub}}'
     test13: '{{hub .ManagedClusterName hub}}'
+    test14: '{{hub (lookup "cluster.open-cluster-management.io/v1" "ManagedCluster" "" .ManagedClusterName).metadata.labels.sites hub}}'
+    test15: '{{hub index ( lookup "cluster.open-cluster-management.io/v1" "ManagedCluster" "" .ManagedClusterName ).metadata.labels "sites" hub}}'
+    test16: '{{hub range $_, $element := (lookup "cluster.open-cluster-management.io/v1" "ManagedCluster" "" "cluster1").status.clusterClaims hub}} {{hub if eq $element.name "id.openshift.io" hub}} {{hub $element.value hub}} {{hub end hub}} {{hub end hub}}'
```
Comment (jc-rh) [inline controllers/utils/policy_template_test.go:57]: Great test :)

File: controllers/utils/policy_template.go [hunk 2]
```diff
@@ -95,6 +103,17 @@ func VerifyHubTemplateFunctions(tmpl interface{}, policyName string) error {
...
 			}
+		} else if hubTmplFunc == "lookup" {
+			matches := regexpLookup.FindStringSubmatch(hubTmpl)
+
+			// Hub template doesn't match the regular expression
+			if len(matches) == 0 {
+				return &PolicyErr{hubTmpl, PlcHubTmplFmtErr}
+			}
+
+			if !strings.HasPrefix(matches[1], "cluster.open-cluster-management.io/") || matches[2] != "ManagedCluster" || matches[3] != "" {
```
Comment (jc-rh) [inline controllers/utils/policy_template.go:114]: are these case-sensitive?

Comment (Missxiaoguo) [reply controllers/utils/policy_template.go:114]: Yes, they are case-sensitive. 
ACM reports the hub template error - one or more API resources are not installed when apiVersion/kind has incorrect capitalization.

**Review by sudomakeinstall2 — APPROVED** [review]

**Pull request merged**